pyop2/global_kernel.py

```diff
@@ -331,14 +333,68 @@ def code_to_compile(self):
         from pyop2.codegen.rep2loopy import generate
 
         wrapper = generate(self.builder)
+        if self._iterset._extruded:
+            iname = "layer"
+        else:
+            iname = "n"
+
+        has_matrix = any(arg._is_mat for arg in self._args)
```

Comment:
I think you want `isinstance(arg, MatKernelArg)` or something similar. I think `_is_mat` is gone.